components/banners.js

```diff
@@ -0,0 +1,70 @@
+import Alert from 'react-bootstrap/Alert'
+import styles from './banners.module.css'
+import { useEffect, useState } from 'react'
+import { useMe } from '../components/me'
+import { useMutation } from '@apollo/client'
+import { WELCOME_BANNER_MUTATION } from '../fragments/users'
+import { useToast } from '../components/toast'
+
+export default function NewVisitorBanner () {
+  const me = useMe()
+  const toaster = useToast()
+  const [hidden, setHidden] = useState(true)
+  const handleClose = async () => {
+    window.localStorage.setItem('hideWelcomeBanner', true)
+    setHidden(true)
+    if (me) {
+      let error
+      try {
+        ({ error } = await hideWelcomeBanner())
+      } catch (e) {
+        error = e
+      }
+      if (error) toaster.danger(error.toString())
+    }
+  }
+  const [hideWelcomeBanner] = useMutation(WELCOME_BANNER_MUTATION, {
+    update (cache) {
+      cache.modify({
+        id: `User:${me.id}`,
+        fields: {
+          hideWelcomeBanner () {
+            return true
+          }
+        }
+      })
+    }
+  })
+  useEffect(() => {
+    setHidden(me?.hideWelcomeBanner || window.localStorage.getItem('hideWelcomeBanner'))
+  }, [me?.hideWelcomeBanner])
+
+  if ((me && !me.hideWelcomeBanner) || !hidden) {
```

Comment:
I may have already asked this but can't find the conversation:

Is `(me && !me.hideWelcomeBanner)` needed anymore?

Since in `setHidden(me?.hideWelcomeBanner || window.localStorage.getItem('hideWelcomeBanner'))`, there already is the whole logic if we should hide or not, no?

If we're logged in, we check `hideWelcomeBanner`. If it's true, we will hide the banner. If it's false, we fallback to local storage which is fine.

If we're not logged in `me?.hideWelcomeBanner` is false and we again fallback to local storage which is fine.

Reply:
It _was_ needed, but with a slight modification to the `setHidden(...)` line it could be eliminated. I also changed the `if` to an early return like in the bookmark banner.